### PR TITLE
[WIP] 0.4.0 beta.5 Server Optional

### LINKED ISSUE #25: Server url from baseUri option
I wonder if it would be possible to make `server` parameter optional? By default we can get url from baseUri option in RAML file.

### PR DESCRIPTION
**Pull request by ProLoser**
Fixes #25 
Makes server optional and defaults to `baseUri`.

Opening this PR so I can comment inline.

### CODE REVIEW

File: test/unit/test-runner-test.coffee [hunk 1]
```diff
@@ -17,7 +17,7 @@ TestRunner = proxyquire '../../lib/test-runner', {
 }
 
 ABAO_IO_SERVER = 'http://abao.io'
-TEST_SERVER = 'http://localhost:3000'
+SERVER = 'http://localhost:3000'
```
Comment (ProLoser) [inline test/unit/test-runner-test.coffee]: Why not move this into the raml file itself

Comment (plroebuck) [inline test/unit/test-runner-test.coffee]: Move what into the RAML file? The `SERVER` variable value? This file was about testing the TestRunner code, not about #25 . Tried to keep most of what was there logically intact. Still not sure about what the `ABAO_IO_SERVER` is supposed to represent as the hostname won't resolve -- or was that the point?

Regardless of this variable, does beta5 meet your needs?

Comment (ProLoser) [inline test/unit/test-runner-test.coffee]: Yes :+1:

**Comment by plroebuck** [comment]
@martnst, apologies for not just asking for PR. Meant to do so earlier and forgot until doing the commit (and just credited you via comment). If you'd like the commit credit to be explicit, send PR for those two modifications in that file and I'll merge yours first.

**Comment by plroebuck** [comment]
@martnst, did you have any problems with the release otherwise?

**Comment by martnst** [comment]
@plroebuck nevermind, i am fine with having the credit comments. 
I am working with our fork due to the need of multiple test cases. I actually forked this other fork https://github.com/lovelybooks/abao (mentioned in #35) which added some temporary hack for multiple test cases.

**Pull request merged**